qprofiler2/src/org/qcmg/qprofiler2/summarise/PairSummary.java

```diff
@@ -0,0 +1,220 @@
+package org.qcmg.qprofiler2.summarise;
+
+import java.util.TreeMap;
+import java.util.concurrent.atomic.AtomicLong;
+import org.qcmg.common.log.QLogger;
+import org.qcmg.common.log.QLoggerFactory;
+import org.qcmg.common.model.QCMGAtomicLongArray;
+import org.qcmg.qprofiler2.util.XmlUtils;
+import org.w3c.dom.Element;
+import htsjdk.samtools.SAMRecord;
+
+public class PairSummary {		
+	public enum Pair{ 
+		F3F5(1), F5F3(2), Inward(3), Outward(4), Others(5); 
+		int id;
+		Pair(int id){this.id = id;}
+	}
+
+	/**
+	 * ignor 
+	 * @param record
+	 * @return
+	 */
+    public static int getOverlapBase(SAMRecord  record) {
+	   	 
+    	//to avoid double counts, we only select one of Pair: tLen > 0 
+        int iSize = record.getInferredInsertSize();
+       	if(iSize < 0) return 0;  
+       	//pick up first of pair if same strand read and tlen=0
+       	if(iSize == 0 && !record.getFirstOfPairFlag() ) return 0;
+        	       	
+       	int result  = 0;  	
+   		//|--> <--| mate end = tLen -1 + read start
+   
+   		if( record.getReadNegativeStrandFlag() != record.getMateNegativeStrandFlag()  ) {  			
+   			// outward pair without overlap
+   			if( iSize == 0) {  return 0;   } 
+   			
+   			//we don't know forward mate end but knows reverse mate end = tLen + read start
+   			int mate_end = iSize + record.getAlignmentStart()-1 ;  //reverse mate end
+   			int read_end = record.getAlignmentEnd();   //forward read start
```

Comment:
camelCase

Reply:
???

Reply:
https://en.wikipedia.org/wiki/Camel_case 
`mateEnd` rather than `mate_end`

Reply:
changed to readEnd